File: scripts/test-wiki-playwright.py
#!/usr/bin/env python3
"""
测试 Playwright 是否能绕过 Cloudflare 访问 Project iM@S Wiki
"""

import asyncio
from playwright.async_api import async_playwright


async def test_wiki():
    async with async_playwright() as p:
        browser = await p.chromium.launch(headless=True)
        context = await browser.new_context(
            user_agent="Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36"
        )
        page = await context.new_page()

        # 访问 Wiki 的 READY!! 页面
        url = "https://project-imas.wiki/api.php?action=parse&page=READY!!&prop=wikitext&format=json"
        print(f"Navigating to: {url}")

        try:
            response = await page.goto(url, wait_until="networkidle", timeout=30000)
            print(f"Response status: {response.status if response else 'None'}")
            print(f"Response URL: {response.url if response else 'None'}")

            content = await page.content()
            # 检查是 JSON 还是 CF challenge HTML
            if content.strip().startswith("{"):
                print("\n✅ SUCCESS: Got JSON response!")
                print(content[:2000])
            elif "Just a moment" in content or "cf-browser-verification" in content:
                print("\n❌ FAILED: Cloudflare challenge page still showing")
                print(content[:500])
            else:
                print("\n⚠️ UNEXPECTED response:")
                print(content[:1000])

        except Exception as e:
            print(f"Error: {e}")

        await browser.close()


if __name__ == "__main__":
    asyncio.run(test_wiki())

File: scripts/test-wiki-playwright.ts
/**
 * Playwright 探针 v3 — 输出实际页面内容
 */
import { chromium } from 'playwright'

async function main() {
  const browser = await chromium.launch({ headless: false })
  const context = await browser.newContext({
    userAgent: 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36',
    viewport: { width: 1280, height: 720 },
  })
  const page = await context.newPage()

  console.log('=== Test: Wiki page (READY!!) ===')
  const wikiUrl = 'https://project-imas.wiki/READY!!'
  try {
    const response = await page.goto(wikiUrl, { waitUntil: 'domcontentloaded', timeout: 60000 })
    console.log('Status:', response?.status())

    // 多等一会儿让 CF challenge 完成
    console.log('Waiting 10s for challenge...')
    await page.waitForTimeout(10000)

    const title = await page.title()
    console.log('Title:', title)

    const bodyText = await page.textContent('body')
    console.log('\n=== Body text (first 600 chars) ===')
    console.log(bodyText?.slice(0, 600))

    // 检查是否有 Wiki 内容特征
    const hasWikiContent = bodyText?.includes('Songbox') || bodyText?.includes('作词') || bodyText?.includes('作曲')
    const hasCF = bodyText?.includes('Just a moment') || bodyText?.includes('请稍候') || bodyText?.includes('Checking your browser')

    console.log('\n=== Detection ===')
    console.log('Has Wiki content:', hasWikiContent)
    console.log('Has CF challenge:', hasCF)

    // 尝试截图
    await page.screenshot({ path: 'scripts/wiki-test-screenshot.png', fullPage: true })
    console.log('Screenshot saved to scripts/wiki-test-screenshot.png')

  } catch (e: any) {
    console.log('Error:', e.message)
  }

  await browser.close()
}

main().catch(console.error)
